ci/jjb/jobs/pulp3_PR_pulp-smash.yaml

```diff
@@ -0,0 +1,256 @@
+
+            source ~/bashrc
+            set -x
+
+            pstop
+            dropdb --user pulp pulp
+            createdb --user pulp pulp
+            workon pulp
+            git remote add pr file://$WORKSPACE/pulp/
+            git fetch pr
+            git checkout pr/pr-branch
+            rm -rf ~/devel/pulp/pulpcore/app/migrations/*
+            preset
+            pstart
+            python manage.py runserver >>~/django_runserver.log 2>&1 &
+
+            mkdir -p ~/.virtualenvs/pulp-smash/
+            python3 -m venv ~/.virtualenvs/pulp-smash
+            source ~/.virtualenvs/pulp-smash/bin/activate
+            pip install --upgrade pip
+            pip install git+https://github.com/PulpQE/pulp-smash.git#egg=pulp-smash
+            mkdir -p ~/.config/pulp_smash
+            cat >~/.config/pulp_smash/settings.json <<EOF
+            {
+                "pulp": {
+                    "auth": ["admin", "admin"],
+                    "version": "3"
+                },
+                "systems": [
+                    {
+                        "hostname": "localhost",
+                        "roles": {
+                            "amqp broker": {"service": "qpidd"},
+                            "api": {"port": 8000, "scheme": "http"},
+                            "mongod": {},
+                            "pulp celerybeat": {},
+                            "pulp cli": {},
+                            "pulp resource manager": {},
+                            "pulp workers": {},
+                            "shell": {},
+                            "squid": {}
+                        }
+                    }
+                ]
+            }
+            EOF
+            pulp-smash settings path
+            pulp-smash settings show
+            pulp-smash settings validate
+            # Use pytest instead of unittest for XML reports. :-(
+            pip install pytest
+            cd $WORKSPACE
+            py.test -v --color=yes --junit-xml=junit-report.xml --pyargs pulp_smash.tests.pulp3
+            test -f junit-report.xml
+            cat ~/django_runserver.log
+
+            echo "Host *
+                StrictHostKeyChecking no" >> ~/.ssh/config
+            sudo chmod 400 ~/.ssh/config
+            export NODE=${JOB_NAME##*/}
+            export SHORT_JOB_NAME=${JOB_NAME/\/$NODE/}
+            curl -k -o $NODE.txt $BUILD_URL/consoleText
+            if grep -q FAILED $NODE.txt ; then mv $NODE.txt $NODE-FAILED.txt; fi
+            rsync -rv --rsync-path="mkdir -p ~/public_html/jenkins/jobs/${SHORT_JOB_NAME}/builds/${BUILD_ID}/ && rsync" $NODE*.txt pulpadmin@repos.fedorapeople.org:public_html/jenkins/jobs/${SHORT_JOB_NAME}/builds/${BUILD_ID}/
+
+    publishers:
+      # Publish a pretty graph of the test results
+      - junit:
+          results: 'junit-report.xml'
+          keep-long-stdio: true
+          test-stability: true
+      - mark-node-offline
+
+- job:
+    name: 'pulp-smash-pulp3-pulp-file-pr'
```

Comment:
How about placing this job in a separate file? For example: `pulp-3-pulp-file-pr`? In the future, the file can be transformed into a template, with jobs named `pulp-3-{plugin}-pr`.